WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -314,17 +314,18 @@ def payment_setup
 
     # Paypal setup URL
     # TODO: Don't generate this URL if there is already a connected paypal account?
-    @paypal_onboarding_url = PaypalInterface.generate_paypal_onboarding_link(@competition.id) unless Rails.env.production?
+    @paypal_onboarding_url = PaypalInterface.generate_paypal_onboarding_link(@competition.id) unless EnvConfig.WCA_LIVE_SITE?
   end
 
   def paypal_return
-    if Rails.env.production?
+    competition = competition_from_params
```

Comment:
You're calling `competition_from_params` here and after the subsequent `if` check, which is unnecessary.